lib/wikidata_sparql2.js

```diff
@@ -117,61 +166,170 @@ module.exports = class WikidataSparqlConverter {
                 outerScope[rhsParam] = rhsScope[rhsParam];
 
             const outerProjection = [];
-            for (let rhsParam of rhsProj)
-                outerProjection.push(rhsParam);
+            for (let rhsParam of rhsProj) outerProjection.push(rhsParam);
 
             for (let lhsParam of lhsProj) {
-                if (outerProjection.indexOf(lhsParam) >= 0)
-                    continue;
+                if (outerProjection.indexOf(lhsParam) >= 0) continue;
                 outerProjection.push(lhsParam);
             }
             return [outerScope, outerProjection];
         } else if (table.isSort) {
-            throw new Error(`todo`);
+            const [scope, projection] = this._tableToSubquery(table.table);
+            let field = table.field;
+            let direction = table.direction;
+            const filterLHS = this._getSparqlVar(scope, field);
+            let sort_code = `ORDER BY ${direction}(${filterLHS})`;
+            this._additionalSparql.push(sort_code);
+            return [scope, projection];
         } else if (Utils.isUnaryTableToTableOp(table)) {
             return this._tableToSubquery(table.table);
         } else {
             throw new TypeError(`Not implemented`);
         }
     }
 
+    reformat_date(date) {
```

Comment:
Starts the function name with "_" if it won't be used outside the class. Or even better, for this one, it can just be a standalone function since it does not touch the class at all.